### PR TITLE
Create a workflow to perform FSim based z-phase calibration

### PR DESCRIPTION
**Pull request by NoureldinYosri**

### CODE REVIEW

**Review by eliottrosenberg — CHANGES_REQUESTED** [review]
This is a great start, Nour!! Thanks for all of your amazing work.

**Comment by eliottrosenberg** [comment]
This should probably go in a separate PR, but I think the next step after this is to write a transformer that replaces the two-qubit gates in a circuit with ones that correct for these z-phases.

File: cirq-core/cirq/experiments/z_phase_calibration.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        - https://arxiv.org/abs/2001.08343
+        - https://arxiv.org/abs/2010.07965
+        - https://arxiv.org/abs/1910.11333
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubits: Qubits to use. If none, use all qubits on the sampler's device.
+        two_qubit_gate: The entangling gate to use.
+        options: The XEB-fitting options. If None, calibrate all 5 PhasedFSimGate parameters,
+            using the representation of a two-qubit gate as an FSimGate for the initial guess.
```
[the remark this answers is not in the document]
Comment (NoureldinYosri) [reply cirq-core/cirq/experiments/z_phase_calibration.py:150]: fixed, it should work now

[the remark this answers is not in the document]
Comment (NoureldinYosri) [reply cirq-core/cirq/experiments/z_phase_calibration.py:150]: btw to speed things up, run the calibration method in a multiprocessing pool e.g.
```py3
with mulitprocessing.Pool(10) as pool:
    result_zphase = cirq.experiments.calibrate_z_phases(sampler, qubits, pool=pool)
```

[the remark this answers is not in the document]
Comment (eliottrosenberg) [reply cirq-core/cirq/experiments/z_phase_calibration.py:150]: Could we make it create a pool by default?

**Comment by NoureldinYosri** [comment]
@mhucka  I already implemented this in a different PR

**Comment by mhucka** [comment]
Ah, it looked like it was closed by the bot due to being stale.

Is the PR #6657?

**Comment by mhucka** [comment]
Then, could you please list the other ones, so that there's a trail for future maintainers?

**Pull request closed without merging**